docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,756 @@
+        subgraph GR-node3
+            rtotr-GR-node3["rtotr-GR-node3 fd97::5/127 fd99::2/64 (0a:58:64:41:00:04)"]
+        end
+    end
+    subgraph layer2-switch
+        stor-ovn_cluster_router["stor-ovn_cluster_router 
+        type: router"]
+    end
+    subgraph ovn_cluster_router["ovn_cluster_router "]
+        trtor-GR-node1["trtor-GR-node1 fd97::8/127"]
+        trtor-GR-node2["trtor-GR-node2 fd97::6/127"]
+        trtor-GR-node3["trtor-GR-node3 fd97::4/127"]
+        rtos-layer2-switch["rtos-layer2-switch 2010:100:200::1 (ipv6-lla: fe80::858:cbff:fecb:1) (0a:58:CB:CB:00:01)"]
+    end
+    rtotr-GR-node1 <--> trtor-GR-node1
+    rtotr-GR-node2 <--> trtor-GR-node2
+    rtotr-GR-node3 <--> trtor-GR-node3
+    VM <-->layer2-switch
+    rtos-layer2-switch <--> stor-ovn_cluster_router
+    
+    class VM vmStyle;
+    class rtotr-GR-node1 portStyle;
+    class rtotr-GR-node2 portStyle;
+    class rtotr-GR-node3 portStyle;
+    class trtor-GR-node1 portStyle;
+    class trtor-GR-node2 portStyle;
+    class trtor-GR-node3 portStyle;
+    class stor-ovn_cluster_router portStyle;
+    class rtos-layer2-switch portStyle;
+    class GR-node1 routerStyle;
+    class GR-node2 routerStyle;
+    class GR-node3 routerStyle;
+    class ovn_cluster_router routerStyle;
+    class layer2-switch switchStyle
+    class term termStyle;
+    class node1,node2,node3 nodeStyle;
+```
+From the IPv6 RFC the gateway address configured from router advertisements is a link local address, in this case it will
+be a link local address computed from the subnbet address `2010:100:200::1`, again this makes the default gw independent of the
+node where the VM is running on.
+
+#### NAT configuration
+As much as possible everything related to conntrack should not be modified since doing so can affect tcp connections, so the 
+only SNAT that need to be moved from GR to ovn_cluster_router is the one enmasquerading the traffic that goes to the mgmt port from the network subnet, since
+now ovn_custer_router is the router connected to the layer2 switch instead of GR, the rest of snat can continue to be done at gateway router.
+```
+llowed_ext_ips     : []
+exempted_ext_ips    : []
+external_ids        : {"k8s.ovn.org/network"=test12_namespace-scoped, "k8s.ovn.org/topology"=layer2}
+external_ip         : "169.254.0.12"
+external_mac        : []
+external_port_range : ""
+gateway_port        : []
+logical_ip          : "203.203.0.0/16"
+logical_port        : rtos-test12_namespace.scoped_ovn_layer2_switch
+match               : "eth.dst == 0a:58:cb:cb:00:02"
+options             : {stateless="false"}
+priority            : 0
+type                : snat
+```
+
+Also the fact that join ip at gateway router is kept allow to maintain all the NATing done with it at gateway router, load balancer for services 
+
+#### Routing an policies
+Moving around routing and policies do not affect tcp connections on upgrade so we can move some routes and policies from gateway router
```

Comment:
Upgrade of a L2 UDN with egress IP is going to break unless you persist the join subnet on the new transit router rtos port. When upgrade rolls, a node is goign to move to this new topology, and another node might still be on the old topology with a route for egress IP to route to this new node. The transit router needs to have the previous join IP on it to avoid disruption.

Reply:
we are keeping the join IP at the gateway router's LRP network, isn't that enough ? or do you mean the additional routes added by egressip ?

Reply:
As we have commented offline, with the current proposal after upgrade a node has no remote LSP to to gateway routers at the switch so moving join IP to rtos is not enough, we were talking about maybe keeping those remove LSPs at the switch and fater upgrade they get removed... but not sure if this make it even more problematic.

Reply:
I think that's fine but we should also keep the old join IP (that used to be on the GR) configured on the TR rtos. The gateway router is not directly connected to the TS anymore after upgrade.

Reply:
@trozet @dceara so what do you think about upgrading node waiting for remote nodes to have the "transit-router-subnet" node annotation before removing logical switch remote LSPs and adding transit router remote LRPs ? I think this will fix the issue

Reply:
Also don't know how much users will have primary layer2 UDNs with egressip when this land, but feels like is going to be near zero.

Reply:
@dceara can we have that just as a temporal mac binding that we remove when the remote port is upgraded too ?

Reply:
@qinqon you'd need a mac binding for each upgraded node on all not-yet-upgraded nodes.  I think it's easier to just configure the old (pre-upgrade) join IP on the TR <-> TS (rtots) link on the upgraded router.

Reply:
We will have to be super careful of generating the mac with the subnet gateway address and not join ip mac.

Reply:
@dceara @trozet I think keepng the remote LSP layer2 swtich is not going to work since the destination mac address from non updated node will be the gateway router and not the ovn_cluster_router so layer2 switch will not match any port.